presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -155,6 +158,29 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         return new HivePartitionResult(partitionColumns, partitionsIterable, compactEffectivePredicate, remainingTupleDomain, enforcedTupleDomain, hiveBucketHandle, bucketFilter);
     }
 
+    public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastore, ConnectorTableHandle tableHandle, List<List<String>> partitionValues)
+    {
+        HiveTableHandle hiveTableHandle = (HiveTableHandle) tableHandle;
+        SchemaTableName tableName = hiveTableHandle.getSchemaTableName();
+
+        Table table = getTable(metastore, tableName);
+        Optional<HiveBucketHandle> hiveBucketHandle = getHiveBucketHandle(table);
+        List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(table);
+        List<Type> partitionTypes = partitionColumns.stream()
+                .map(column -> typeManager.getType(column.getTypeSignature()))
+                .collect(toList());
+
+        Iterable<HivePartition> partitionsIterable = () -> partitionValues.stream()
+                .map(partitionValue -> getPartitionNamesByParts(metastore, tableName, partitionValue))
+                .flatMap(partitionNames -> partitionNames.stream()
+                        .map(partitionName -> parseValuesAndFilterPartition(tableName, partitionName, partitionColumns, partitionTypes, alwaysTrue())))
+                .filter(Optional::isPresent)
```

Comment:
Don't filter, but verify. It should never return `empty`, since you are passing `alwaysTrue()` domain